patroni/postgresql.py

```diff
@@ -567,10 +581,51 @@ def time_in_state(self):
     def is_starting(self):
         return self.state == 'starting'
 
+    def wait_for_port_open(self, timeout, initiated, completed):
+        """Waits until PostgreSQL opens ports."""
+        for _ in polling_loop(timeout):
+            pid_file = self.read_pid_file()
+            if len(pid_file) < 5:
+                # Pid file doesn't exist or is incomplete, wait for it to be created
+                continue
+            try:
+                pid = int(pid_file['pid'])
+                postmaster_start_time = int(pid_file['start_time'])
+            except ValueError:
+                # Garbage in the pid file
+                continue
+
+            timing_slop = 2
+            is_our_pid_file = (initiated - timing_slop) <= postmaster_start_time <= (completed + timing_slop)
+            if not self.is_pid_running(pid):
+                if is_our_pid_file:
+                    # Start failed
+                    return False
+                else:
+                    # Probably old pid file, need to wait for cleanup
```

Comment:
About that, why is that necessary? Doesn't seem like it would be safe. I poked around in git history and that code is relatively ancient with a very uninformative commit message.